src/slider/interfaces.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { BaseComponentProps } from '../internal/base-component';
+import { FormFieldValidationControlProps } from '../internal/context/form-field-context';
+import { NonCancelableEventHandler } from '../internal/events';
+
+export interface SliderProps extends BaseComponentProps, FormFieldValidationControlProps {
+  /**
+   * Indicates the current value.
+   */
+  value?: number;
+
+  /**
+   * Indicates the minimum value.
+   */
+  min: number;
+
+  /**
+   * Indicates the maximum value.
+   */
+  max: number;
+
+  /**
+   * Formats the values. This will format both the labels and the tooltip.
+   */
+  valueFormatter?: (value: number) => string;
+
+  /**
+   * onChange handler.
+   */
+  onChange?: NonCancelableEventHandler<SliderProps.ChangeDetail>;
+
+  /**
+   * How big the step size is.
+   */
+  step?: number;
```

Comment:
What happens to the component if step is greater than (max - min)?

Reply:
It should not be valid. Shall we throw a warning?

Reply:
I think a warning and some fallback would do. Not sure what can be the fallback exactly. Maybe we can renderd the component but in a disabled state or something

Reply:
I think we just do what the HTML reference does, which is a slider that cannot be moved. That plus a warnOnce makes sense to me.